spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1505,19 +1508,23 @@ def create_new_client(self, give_focus=True, filename='', is_cython=False,
             has_spyder_kernels = programs.is_module_installed(
                 'spyder_kernels',
                 interpreter=pyexec,
-                version='>=2.2.0;<2.3.0')
+                version=SPYDER_KERNELS_VERSION)
             if not has_spyder_kernels and not running_under_pytest():
                 client.show_kernel_error(
                     _("Your Python environment or installation doesn't have "
                       "the <tt>spyder-kernels</tt> module or the right "
-                      "version of it installed (>= 2.2.0 and < 2.3.0). "
+                      "version of it installed ({0}). "
                       "Without this module is not possible for Spyder to "
                       "create a console for you.<br><br>"
                       "You can install it by running in a system terminal:"
                       "<br><br>"
-                      "<tt>conda install spyder-kernels=2.1</tt>"
+                      "<tt>{1}</tt>"
                       "<br><br>or<br><br>"
-                      "<tt>pip install spyder-kernels==2.1.*</tt>")
+                      "<tt>{2}</tt>".format(
+                          SPYDER_KERNELS_VERSION,
```

Comment:
It's a great idea to use constants in this message! I suggest to introduce the following ones:

```python
SPYDER_KERNELS_MIN_VERSION = '2.2.0'
SPYDER_KERNELS_MAX_VERSION = '2.3.0'
```

which will allow us to write the others as

```python
SPYDER_KERNELS_VERSION = (
    f'>={SPYDER_KERNELS_MIN_VERSION};<SPYDER_KERNELS_MAX_VERSION')
SPYDER_KERNELS_VERSION_MSG = (
    f'>= {SPYDER_KERNELS_MIN_VERSION} and < SPYDER_KERNELS_MAX_VERSION')
SPYDER_KERNELS_CONDA = (
    f'conda install spyder-kernels={SPYDER_KERNELS_MIN_VERSION[:-2]}')
SPYDER_KERNELS_PIP = (
    f'pip install spyder-kernels=={SPYDER_KERNELS_MIN_VERSION[:-1]}*')
```

That way, we could use here `SPYDER_KERNELS_VERSION_MSG` instead of `SPYDER_KERNELS_VERSION`. Also, the only two constants that we should modify are `SPYDER_KERNELS_MIN_VERSION` and `SPYDER_KERNELS_MAX_VERSION`, which reduces the possibility of making mistakes.